### PR TITLE
feat: add default delete condition to managed dependent resources

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

**Review by stevekuznetsov — COMMENTED** [review]
Are you able to use ObjectReferences to allow k8s to do the cascading deletion for you?

**Comment by csviri** [comment]
Hi @stevekuznetsov , KubernetesDependentResource handle owner references by default for you, this can be cusotmized also in multiple ways. See: https://javaoperatorsdk.io/docs/dependent-resources#telling-josdk-how-to-find-which-secondary-resources-are-associated-with-a-given-primary-resource

See also [`GarbageCollected`](
https://github.com/java-operator-sdk/java-operator-sdk/blob/16066ea738a33a9577bf84bc07a82a9a2e01582a/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/GarbageCollected.java#L28-L28) interface docs.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/KubernetesResourceDeletedCondition.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package io.javaoperatorsdk.operator.processing.dependent.workflow;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
+
+public class KubernetesResourceDeletedCondition implements Condition<HasMetadata, HasMetadata> {
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/KubernetesResourceDeletedCondition.java:7]: This needs to be better documented.

File: docs/documentation/workflows.md [hunk 2]
```diff
@@ -312,6 +312,21 @@ containing all the related exceptions.
 The exceptions can be handled
 by [`ErrorStatusHandler`](https://github.com/java-operator-sdk/java-operator-sdk/blob/14620657fcacc8254bb96b4293eded84c20ba685/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/ErrorStatusHandler.java)
 
+## Notes on Ordered Kubernetes Resource Cleanups
+
+In case there are Kubernetes Dependent Resources that are depend on each other, and are ordered -
+like when those represent some external resources and would be nice to have them created and deleted in order -
+on cleanup the resources will be deleted in reverse order. However, the workflow the implementation currently just
+calls delete on resource, in case it uses a finalizer, it should be checked by a `DeletePostCondition` if it
```
Comment (metacosm) [inline docs/documentation/workflows.md:320]: This isn't really clear and should be better explained, as it's not clear why having a finalizer changes things and why a delete post condition is needed.

**Review by metacosm — CHANGES_REQUESTED** [review]

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/KubernetesResourceDeletedCondition.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/KubernetesResourceDeletedCondition.java:12]: added javadoc

**Pull request merged**